### PR TITLE
Add openapi endpoint

### PR DESCRIPTION
**Pull request by thenav56**
## Changes
Add base open API v3 endpoint
- Using [drf-spectacular](https://drf-spectacular.readthedocs.io/en/latest/) for schema generation
- Endpoint:
  - http://localhost:8000/api-docs/ -> YAML
  - http://localhost:8000/api-docs/swagger-ui/
  - http://localhost:8000/api-docs/redoc/
- Replace existing /docs/ with redoc ui
  
> NOTE: Make sure to build a docker image

```bash
docker-compose exec serve ./manage.py spectacular --color
docker-compose exec serve ./manage.py spectacular --color --file openapi-v3.yml
```
```
Schema generation summary:
Warnings: 109 (89 unique)
Errors:   224 (50 unique)
```

### CODE REVIEW

File: main/urls.py [hunk 1]
```diff
@@ -168,6 +171,17 @@
 
 router.register(r"review-country", api_views.CountryOfFieldReportToReviewViewset, basename="review_country")
 
+api_schema_view = get_schema_view(
+    openapi.Info(
+        title="DEEP API",
+        default_version='v1',
+        description="DEEP API",
+        contact=openapi.Contact(email="admin@thedeep.io"),
```
Comment (frozenhelium) [inline main/urls.py:179]: Let's update these info for IFRC

**Review by frozenhelium — CHANGES_REQUESTED** [review]

File: main/urls.py (hunk 1 above)
Comment (thenav56) [reply main/urls.py]: Done

**Comment by tnagorra** [comment]
## OpenAPI v3

We have updated the OpenAPI to version 3.
It includes a command to show all of the errors/warning related to schema

Here is an output of `docker-compose exec serve ./manage.py spectacular --color`

image

## Redoc

We are now using Redoc to generate documentation for http://localhost:8000/docs/
There might be minor differences from the previous generated documentation.

## Next Steps

- Fix the errors/warning listed gradually.
- Check if the generated types are correct for all of the endpoints.

@batpad @szabozoltan69 Can we also discuss on how we can continue this work and improve the generated OpenAPI types.

**Comment by frozenhelium** [comment]
@szabozoltan69 please review and merge this

**Review by frozenhelium — APPROVED** [review]

**Review by szabozoltan69 — APPROVED** [review]

**Comment by batpad** [comment]
Spectacular work, @thenav56 @tnagorra ! Let's discuss how we can fix up errors on the dev call or so.

**Comment by szabozoltan69** [comment]
Merged to develop and deployed to Staging. (Error fixing later)

**Comment by szabozoltan69** [comment]
kép

**Comment by szabozoltan69** [comment]
@thenav56 @tnagorra is it possible to use this /docs endpoint interactively? So, giving some arguments and sending the request?

**Comment by szabozoltan69** [comment]
Or maybe could we keep the old possibility as a /docs_v1 ?

**Comment by thenav56** [comment]
@szabozoltan69 swagger-ui has interaction, maybe we can use that http://localhost:8000/api-docs/swagger-ui/

**Comment by tnagorra** [comment]
@szabozoltan69 I also suggest we use https://goadmin-stage.ifrc.org/api-docs/swagger-ui/ for interactivity.

**Comment by szabozoltan69** [comment]
I am not sure that this kind of output structure could remain, which is in https://goadmin-stage.ifrc.org/api/v2/brief
It's just a healthcheck endpoint, for human reading, not for any other machine usage.
(It's the only one which is "hacked" like this.)

**Comment by thenav56** [comment]
I think that should be okay. Basically, we define a serializer that we use to generate the data instead of a manual response. Then the schema will show the structure.

**Pull request merged**